services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/FetchCredentialsExecutionService.java

```diff
@@ -0,0 +1,83 @@
+package com.bigid.azurekeyvaultapp.service.impl;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.security.keyvault.secrets.SecretClient;
+import com.azure.security.keyvault.secrets.SecretClientBuilder;
+import com.bigid.appinfrastructure.dto.ExecutionContext;
+import com.bigid.appinfrastructure.dto.StatusEnum;
+import com.bigid.appinfrastructure.externalconnections.BigIDProxy;
+import com.bigid.appinfrastructure.services.AbstractExecutionService;
+import com.bigid.azurekeyvaultapp.constant.ActionParams;
+import com.bigid.azurekeyvaultapp.service.ExecutionService;
+import com.bigid.azurekeyvaultapp.service.KeyVaultTokenService;
+import com.bigid.azurekeyvaultapp.util.ParamsMapUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Triple;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+
+@Service
+@Slf4j
+public class FetchCredentialsExecutionService extends AbstractExecutionService implements ExecutionService {
+
+    private static final String SECRET_KEY = "secret_key";
+    private static final String SUCCESSFULLY_FETCHED_SECRET = "Successfully fetched secret";
+    public static final String FAILED_TO_FETCH_SECRET = "Failed to fetch secret";
+
+    private final KeyVaultTokenService keyVaultTokenService;
+
+    @Autowired
+    public FetchCredentialsExecutionService(BigIDProxy bigIDProxy, KeyVaultTokenService keyVaultTokenService) {
+        super(bigIDProxy);
+        this.keyVaultTokenService = keyVaultTokenService;
+    }
+
+    @Override
+    public Triple<Boolean, String, Map<String, String>> performAction(ExecutionContext executionContext) {
```

Comment:
Might be nice to create a AzureVaultResponseClass with overrides for getLeft, getMiddle, getRight with the appropriate names for the fields to help with readability in the controller